### ISSUE TITLE
Add resolvers parameter to Provider class

### ISSUE THREAD

**Issue by linasi** [RadicalLedger/did-siop-lib#24]
Both `RP` and `Provider` classes resolve DIDs. `RP` has a parameter to add additional custom resolvers:
```
getRP(redirect_uri: string, did: string, registration: any, did_doc?: DidDocument, **resolvers?:DidResolver[]**,op_metadata?:any)
```
while `Provider` is missing such functionality.

`Provider` should also allow to configure resolvers via constructor or some additional instance method.

**Comment by fidenz-chim** [comment]
Fixed in  [b2be84b](https://github.com/RadicalLedger/did-siop-lib/tree/b2be84b00c72adb2fb12a343e3f1dec48f1656ba)

**Comment by linasi** [comment]
Thank you @fidenz-chim!

Do you think that `did_doc?: DidDocument` argument is still necessary in `getRP` and `getProvider` functions? If it's passed - resolvers argument will be ignored. In Provider case one hardcoded DID will never be enough anyway.
